texar/modules/decoders/rnn_decoders.py

```diff
@@ -486,6 +486,34 @@ def default_hparams():
         }
         return hparams
 
+    def initialize(  # type: ignore
+            self, helper: Helper,
+            inputs: Optional[torch.Tensor],
+            sequence_length: Optional[torch.LongTensor],
+            initial_state: Optional[MaybeList[MaybeTuple[torch.Tensor]]]) -> \
+            Tuple[torch.ByteTensor, torch.Tensor,
+                  Optional[AttentionWrapperState]]:
+        initial_finished, initial_inputs = helper.initialize(inputs,
+                                                             sequence_length)
+        if initial_state is None:
+            state = None
+        else:
+            if isinstance(initial_state, list):
+                if isinstance(initial_state[0], tuple):
+                    batch_size = initial_state[0][0].shape[0]
+                else:
+                    batch_size = initial_state[0].shape[0]
+            else:
+                if isinstance(initial_state, tuple):
+                    batch_size = initial_state[0].shape[0]
+                else:
+                    batch_size = initial_state.shape[0]
```

Comment:
I actually suggest using `map_structure` for this. It could be done as:
```python
batch_size = None

def _get_batch_size_fn(x):
	nonlocal batch_size
	if isinstance(x, torch.Tensor):
		batch_size = x.size(0)

map_structure(_get_batch_size_fn, initial_state)
```